### PR TITLE
CM-229 Implemented tasks for CRUD operations

### PR DESCRIPTION
**Pull request by malinowskikam**
[https://openimis.atlassian.net/browse/CM-229](https://openimis.atlassian.net/browse/CM-229)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,7 +58,9 @@ It is dedicated to be deployed as a module of [openimis-be_py](https://github.co
 * gql_beneficiary_update_perms: required rights to call updateBeneficiary and updateGroupBeneficiary GraphQL Mutation (default: ["160003"])
 * gql_beneficiary_delete_perms: required rights to call deleteBeneficiary and deleteGroupBeneficiary GraphQL Mutation (default: ["170004"])
 
-* check_benefit_plan_update: specifies whether Benefit Plan update should be approved (default: True)
+* gel_check_benefit_plan_update: specifies whether Benefit Plan update should be approved (default: True)
+* gql_check_beneficiary_crud: specifies whether Beneficiary should should be approved (default: True)
+* gql_check_group_beneficiary_crud: specifies whether Group Beneficiary should should be approved (default: True),
```
Comment (dborowiecki) [inline README.md:63]: I think that we should have unified semantics for this (currently we maker checker logic, tasks based approval and so on), but for now I think not specific enough. For now I think we could go for
```suggestion
* gql_check_benefit_plan_update: specifies whether Benefit Plan update should be updated using task based approval (default: True)
* gql_check_beneficiary_crud: specifies whether Beneficiary CRUD should be use task based approval (default: True)
* gql_check_group_beneficiary_crud: specifies whether Group Beneficiary should use tasks based approval (default: True),
```

File: social_protection/gql_mutations.py [hunk 2]
```diff
@@ -178,12 +178,16 @@ def _mutate(cls, user, **data):
...
         if ids:
             with transaction.atomic():
-                for id in ids:
-                    service.delete({'id': id})
+                for obj_id in ids:
+                    res = service.delete({'id': obj_id})
+                    if not res['success']:
+                        transaction.rollback()
+                        break
+        return res if not res['success'] else None
```
Comment (dborowiecki) [inline social_protection/gql_mutations.py:190]: Thanks for this improvement. In terms of code I think we could reduce the indentation with something like: 
```python
    ids = data.get('ids')
    if not ids:
        return {'success': False, 'message': 'No IDs to delete', 'details': ''}

    service = BenefitPlanService(user)
    with transaction.atomic():
        for obj_id in ids:
            res = service.delete({'id': obj_id})
            if not res['success']:
                transaction.rollback()
                return res

    return None
```
let me know your thoughts.

Comment (dborowiecki) [reply social_protection/gql_mutations.py:190]: Same would apply for remaining mutations that were changed.

**Review by dborowiecki — CHANGES_REQUESTED** [review]
I left small improvement ideas.

File: social_protection/gql_mutations.py (hunk 2 above)
Comment (malinowskikam) [reply social_protection/gql_mutations.py:190]: I like this version more

**Comment by malinowskikam** [comment]
This PR depends on https://github.com/openimis/openimis-be-tasks_management_py/pull/15

**Review by dborowiecki — APPROVED** [review]

**Pull request merged**